presto-main/src/main/java/com/facebook/presto/sql/gen/BytecodeUtils.java

```diff
@@ -181,45 +180,86 @@ public static BytecodeNode generateInvocation(Scope scope, String name, ScalarFu
         // Index of parameter (without @IsNull) in Presto function
         int realParameterIndex = 0;
 
+        // Go through all the choices in the function and then pick the best one
+        List<ScalarImplementationChoice> choices = function.getAllChoices();
+        ScalarImplementationChoice bestChoice = null;
+        for (int i = 0; i < choices.size(); i++) {
+            boolean isValid = true;
+            ScalarImplementationChoice current = choices.get(i);
+            for (int j = 0; j < arguments.size(); j++) {
+                if (current.getArgumentProperty(j).getArgumentType() != VALUE_TYPE) {
+                    continue;
+                }
+                if (!(arguments.get(j) instanceof InputReferenceNode) && current.getArgumentProperty(j).getNullConvention() == NullConvention.BLOCK_AND_POSITION) {
+                    isValid = false;
+                    break;
+                }
+            }
+            if (isValid) {
+                bestChoice = choices.get(i);
+            }
+        }
+
+        checkState(bestChoice != null);
```

Comment:
`checkState(bestChoice != null, "None of the scalar function implementation choices are valid");`